### PR TITLE
Mentors

### PR DESCRIPTION
**Pull request by ghost**
:cl: More Robust Than You
server: Added mentors.
/:cl:

They're kinda like admins, but instead of admining and moderating, they babysit retards through game mechanics.

### CODE REVIEW

**Comment by lzimann** [comment]
this is useless

**Comment by Potato-Masher** [comment]
I have always kinda liked the idea of a mentor system. Keeps the new players from bothering the admins with pointless ahelps that could be answered by another player, and keeps them from accidentally doing an "ICKY OCCY" moment and getting everyone pissed off.

**Comment by Jordie0608** [comment]
We're not bothered by players ahelping questions and this is all but duplicating existing admin code to accomplish a similar system that could be handled without copypasta.

Doesn't seem like a good idea.

**Comment by awesine** [comment]
good idea
mentoring ABSOLUTELY helped me get into ss13 on the first server I played, goonstation.
it's helped a lot of people there and it'll probably help people here

**Comment by deathride58** [comment]
Yo

This would be pretty useful for Citadel and Hippie, as both servers need a functioning mentor system. Having the system implemented here would prevent a lot of headaches related to maintaining the feature on both servers, and would make it easier for other downstreams in the future to use the system.

TG doesn't have to use the feature itself, as server and codebase separation is a thing here.

**Comment by SamHPurp** [comment]
> would make it easier for other downstreams in the future to use the system.

Yessir

**Comment by awesine** [comment]
if this gets merged open up a mentor applications on the forum

**Comment by Bawhoppen** [comment]
We've already tried to add mentors before. They were shot down. Obviously this would have to go hand in hand with policy support, which I don't think you really have.

**Comment by FantasticFwoosh** [comment]
Id volunteer as a mentor, you can talk shit about me off game but i know plenty of handy things and can learn more. Technically wouldn't violate my self commitment either as i wouldn't be a active participant.

👍  I'd say give it a spin.

**Comment by XDTM** [comment]
The reason i like mentors is that i don't play nearly enough to be an admin, but i like giving advice to players and would happily volunteer for mentoring

**Comment by kevinz000** [comment]
why the hell are you making ANOTHER ADMIN SYSTEM JUST FOR MENTORS WHEN WE HAVE ONE PERFECTLY CAPABLE OF HANDLING RANKS AND PERMISSIONS?

File: code/modules/mentor/mentor_verbs.dm [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+	if(key)
+		if(include_link)
+			if(CONFIG_GET(flag/mentors_mobname_only))
+				. += "<a href='?mentor_msg=[REF(M)]'>"
+			else
+				. += "<a href='?mentor_msg=[ckey]'>"
+
+		if(C && C.holder && C.holder.fakekey)
+			. += "Administrator"
```
Comment (kevinz000) [inline code/modules/mentor/mentor_verbs.dm]: Bullshit.

**Comment by kevinz000** [comment]
Also if this ridiculous PR goes through and yuo don't add a system for people with admin +PERMISSIONS to add and remove mentors and their abilities I'm going to be really upset.

File: code/modules/mentor/mentor_verbs.dm (hunk 1 above)
Comment (kevinz000) [reply code/modules/mentor/mentor_verbs.dm]: >fakekey

the amoutn of copypaste is real.

**Comment by lzimann** [comment]
This is just a copypasta, not only the admin holder but also a direct copy from hippie, you haven't even bothered compiling this locally. If you truly want to add mentors, then you will have to do this whole thing properly, using the existing permission system, and tweak the existing code to support mentors that won't have any admin priviledge, adding yet another holder that looks exact the same when all of this can be achieved with what we already have is a big no.

As to bothering admins, players ahelping questions about the game has never bothered any admin, nor admins are super busy to need mentors. 

This will be closed, open a new pr if you actually do it anything so we can have a clean start.

**Comment by duncathan** [comment]
there's a 0% chance this PR goes through as is
if the code were decent I'd be up for including this for downstreams but as is this is garbage

**Comment by duncathan** [comment]
if anyone spouts "muh 24 hours" I will publically make fun of you in coderbus

**Comment by evsey9** [comment]
"As to bothering admins, players ahelping questions about the game has never bothered any admin, nor admins are super busy to need mentors."
i ahelped "whats the throwforce of %something%" a day ago and literally got told to look in the code

**Comment by Cyberboss** [comment]
Let me be very clear. This code as is is absolutely terrible and it will never get in on my watch

**Comment by Bawhoppen** [comment]
Mentor system wouldn't be bad, this is just half-assed version

**Comment by MoreRobustThanYou** [comment]
actually this code was an unholy yog-hippie hybrid

**Comment by epochayur** [comment]
I really, really want a mentor system. Speaking from personal experience, the retention of new folks is much greater with real people showing them around and talking them through the interface as opposed to reading a dozen different wiki pages. Not saying the wiki is useless, it's just a much smoother transition period when there's someone there to show them the ropes. New blood is essential to keeping the community going, and I think it's unfortunate that a lot of people drop it in the first few minutes because they don't really understand what's happening, and I think a mentor system would help a lot. Code aside, are people against the concept in principle?

**Comment by Bawhoppen** [comment]
We just need to make it clear to people they shouldn't be afraid to ahelp any questions they have. Not sure how to implement that though

**Pull request closed without merging**